### ISSUE TITLE
Cron job produces endless error message loop

### ISSUE THREAD

**Issue by MichaIng** [doganoo/RecommendationAssistant#4]
### Steps to reproduce
1. Install the app
2. Wait until first cron job starts the apps file scan (?)
3. Watch Nextcloud log entries

### Expected behaviour
No errors should show up

### Actual behaviour

Level | App | Message
-- | -- | --
Error | RecommendationAssistant | the similarity value has to be between 0 and 5, NAN given
Warning | RecommendationAssistant | [2,"Division by zero","\/var\/www\/nextcloud\/apps\/recommendation_assistant\/lib\/Recommendation\/CosineComputer.php",99]
- Those error messages appear every few seconds, repeatedly, while the cron job is running, now for 7.5 hours.

### Server configuration
- Issue template app not yet available for NC14

**Operating system**: Raspbian Buster

**Web server:** Apache2 with mod_php

**Database:** MariaDB 10.1

**PHP version:** 7.2

**Nextcloud version:** 14 Beta 4

**Where did you install Nextcloud from:** https://download.nextcloud.com/server/

**List of activated apps:**

```
Enabled:
  - activity: 2.7.0
  - apporder: 0.5.0
  - calendar: 1.6.1
  - cloud_federation_api: 0.0.1
  - contacts: 2.1.6
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - files: 1.9.0
  - files_downloadactivity: 1.3.0
  - files_rightclick: 0.8.4
  - files_sharing: 1.6.2
  - files_trackdownloads: 1.2.2
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - gallery: 18.1.0
  - impersonate: 1.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - notes: 2.4.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - previewgenerator: 1.1.0
  - provisioning_api: 1.4.0
  - serverinfo: 1.4.0
  - spreed: 3.99.10
  - twofactor_backupcodes: 1.3.0
  - updatenotification: 1.4.1
  - workflowengine: 1.4.0
Disabled:
  - accessibility
  - admin_audit
  - comments
  - encryption
  - federation
  - files_external
  - files_pdfviewer
  - files_texteditor
  - files_videoplayer
  - firstrunwizard
  - nextcloud_announcements
  - password_policy
  - polls
  - ransomware_protection
  - recommendation_assistant
  - sharebymail
  - sharerenamer
  - support
  - survey_client
  - systemtags
  - tasks
  - theming
  - user_external
  - user_ldap

```

**Nextcloud configuration:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "my.domain.org"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "14.0.0.16",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": true,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 1,
        "logtimezone": "Europe\/Berlin",
        "trashbin_retention_obligation": "disabled",
        "versions_retention_obligation": "disabled",
        "skeletondirectory": "",
        "defaultapp": "apporder",
        "maintenance": false,
        "overwrite.cli.url": "https:\/\/localhost\/nextcloud",
        "htaccess.RewriteBase": "\/nextcloud",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "mysql.utf8mb4": true,
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

#### Nextcloud log (data/nextcloud.log)

Level | App | Message
-- | -- | --
Error | RecommendationAssistant | the similarity value has to be between 0 and 5, NAN given
Warning | RecommendationAssistant | [2,"Division by zero","\/var\/www\/nextcloud\/apps\/recommendation_assistant\/lib\/Recommendation\/CosineComputer.php",99]

**Comment by doganoo** [comment]
@MichaIng can you please update the app and check again?

**Comment by MichaIng** [comment]
@doganoo 
Jep, issue resolved with v0.2.2 👍

**Comment by doganoo** [comment]
@MichaIng would love to get your feedback about recommended files (if any)

**Comment by MichaIng** [comment]
@doganoo
Sorry the issue is indeed not solved. Same error messages appear after obviously not every (15 minutes) cron job execution. But hang cron jobs added up to three that I needed to kill manually.

I guess your apps background job is execute every several hours instead?

**Comment by doganoo** [comment]
@MichaIng the normal behaviour is the execution in a defined time interval. But I thought I even have fixed the "division by zero" bug. I'll have a closer look...

**Comment by MichaIng** [comment]
:+1: Just tell me, if I can help with some debugging here.

**Comment by doganoo** [comment]
@MichaIng could you please post the exact error message?

**Comment by doganoo** [comment]
I have released a new version (1.0.2). Please update and tell me if this fixes this issue.
